src/tribol/interface/tribol.cpp

```diff
@@ -391,6 +408,107 @@ void registerMesh( integer meshId,
+                      integer contact_case,
+                      integer contact_method,
+                      integer contact_model,
+                      integer enforcement_method,
+                      integer binning_method)
+{
+   std::unique_ptr<mfem::FiniteElementCollection> dual_fec = nullptr;
+   integer dual_vdim = 0;
+   auto mfem_data = std::make_unique<MfemMeshData>(
+      mesh_id_1,
+      mesh_id_2,
+      mesh,
+      current_coords,
+      attributes_1,
+      attributes_2
+   );
+   // register empty meshes so the coupling scheme is valid
+   registerMesh(
+      mesh_id_1, 0, 0, nullptr, mfem_data->GetElemType(), nullptr, nullptr);
+   registerMesh(
+      mesh_id_2, 0, 0, nullptr, mfem_data->GetElemType(), nullptr, nullptr);
+   registerCouplingScheme(
+      cs_id,
+      mesh_id_1,
+      mesh_id_2,
+      contact_mode,
+      contact_case,
+      contact_method,
+      contact_model,
+      enforcement_method,
+      binning_method
+   );
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   if (enforcement_method == LAGRANGE_MULTIPLIER)
+   {
+      dual_fec = std::make_unique<mfem::H1_FECollection>(
+         current_coords.FESpace()->FEColl()->GetOrder(),
+         mesh.SpaceDimension()
+      );
+      if (contact_model == FRICTIONLESS)
+      {
+         dual_vdim = 1;
+      }
+      else if (contact_model == TIED || contact_model == COULOMB)
+      {
+         dual_vdim = mesh.SpaceDimension();
+      }
+      else
+      {
+         SLIC_ERROR_ROOT("Unsupported contact model. "
+           "Only FRICTIONLESS, TIED, and COULOMB supported.");
+      }
+      coupling_scheme->setMfemSubmeshData(
+         std::make_unique<MfemSubmeshData>(
+            mfem_data->GetSubmesh(),
+            mfem_data->GetLORSubmesh(),
+            std::move(dual_fec),
+            dual_vdim
+         )
+      );
+      auto lm_options = coupling_scheme->getEnforcementOptions().lm_implicit_options;
+      if (
+         lm_options.enforcement_option_set && 
+         (
+            lm_options.eval_mode == ImplicitEvalMode::MORTAR_JACOBIAN ||
+            lm_options.eval_mode == ImplicitEvalMode::MORTAR_RESIDUAL_JACOBIAN
+         )
+      )
+      {
+         coupling_scheme->setMatrixXfer(std::make_unique<MfemMatrixData>(
+            *mfem_data,
+            *coupling_scheme->getMfemSubmeshData()
+         ));
+      }
+   }
+   coupling_scheme->setMfemMeshData(std::move(mfem_data));
+
+} // end of registerParMesh()
+
+//------------------------------------------------------------------------------
+void setMfemLowOrderRefinedFactor( integer cs_id,
+                                   integer lor_factor)
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemData(),
+      "Coupling scheme does not contain MFEM data. "
+      "Was the coupling scheme created with registerParMesh()?"
```

Comment:
registerParMesh() is not what it is called right now, and we may change it. Just update to whatever we go with. Change to "Coupling scheme does not contain MFEM data. Call <insert function name here> to use low order refined feature." (or something like that to avoid rhetorical questions being posed to user :) )

Reply:
Thanks for noticing, it's been fixed.